spec/requests/controllers_specs_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+RSpec.describe 'ControllersSpecs', type: :request do
+  describe 'UsersController' do
+    it 'renders the index page' do
+      get '/users'
+      expect(response).to have_http_status(200)
+      expect(response).to render_template(:index)
+      expect(response.body).to include('All Users Page')
+    end
+    it 'renders the show page' do
+      user = create(:user)
+      get "/users/#{user.id}"
+      expect(response).to have_http_status(200)
+      expect(response).to render_template(:show)
+      expect(response.body).to include('Single User Page')
+    end
+  end
+  describe 'PostsController' do
+    it 'renders the posts index page for a user' do
+      user = create(:user)
+      get "/users/#{user.id}/posts"
+      expect(response).to have_http_status(200)
+      expect(response).to render_template(:index)
+      expect(response.body).to include('All Posts Page')
+    end
+    it 'renders the posts show page for a selected post' do
+      user = create(:user) # Assuming you have FactoryBot configured
+      post = create(:post, author: user)
+      get "/users/#{user.id}/posts/#{post.id}"
+      expect(response).to have_http_status(200)
+      expect(response).to render_template(:show)
+      expect(response.body).to include('Single Post Page')
+    end
+  end
```

Comment:
- [ ] Please make sure to place the test in different files for users and posts controllers as requested in the project requirements.
![image](https://github.com/demesameneshoa/blog-app/assets/83803414/060eac7c-c055-4308-a4cc-c6d8693ffcdb)